stress/shell/src/main/java/alluxio/stress/cli/MultiOperationStressMasterBench.java

```diff
@@ -0,0 +1,479 @@
+          break;
+        }
+
+        long operationCount;
+        operationIndex = operationIndexSelector.apply(operationIndex);
+        operationCount = mContext.getOperationCounter(operationIndex).incrementAndGet() - 1;
+        long startNs = System.nanoTime();
+        applyOperation(operationIndex, operationCount);
+        long endNs = System.nanoTime();
+
+        long currentMs = CommonUtils.getCurrentMs();
+        // Start recording after the warmup
+        if (currentMs > recordMs) {
+          mResult.incrementNumSuccess(operationIndex, 1);
+
+          // record response times
+          long responseTimeNs = endNs - startNs;
+          mResponseTimeNs[operationIndex].recordValue(responseTimeNs);
+
+          // track max response time
+          long[] maxResponseTimeNs = mResult.getStatistics(operationIndex).mMaxResponseTimeNs;
+          int bucket =
+              Math.min(maxResponseTimeNs.length - 1, (int) ((currentMs - recordMs) / bucketSize));
+          if (responseTimeNs > maxResponseTimeNs[bucket]) {
+            maxResponseTimeNs[bucket] = responseTimeNs;
+          }
+        }
+      }
+    }
+
+    //Each thread will randomly pick an operation to do by operations ratio.
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runOperationsRatioBased() throws Exception {
+      double ratioSum = 0;
+      double last = 0;
+      double[] ratioArray = new double[mParameters.mOperations.length];
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioSum += mParameters.mOperationsRatio[i];
+      }
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioArray[i] = last + mParameters.mOperationsRatio[i] / ratioSum;
+        last = ratioArray[i];
+      }
+      runInternal(lastOperationIndex -> {
+        double r = mRandom.nextDouble();
+        int operationIndex = 0;
+        for (int i = 0; i < mParameters.mOperations.length; i++) {
+          if (ratioArray[i] >= r) {
+            operationIndex = i;
+            break;
+          }
+        }
+        mContext.getRateLimiter(0).acquire();
+        return operationIndex;
+      });
+    }
+
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runThreadsRatioBased() throws Exception {
+      double ratioSum = 0;
+      double last = 0;
+      double[] ratioArray = new double[mParameters.mOperations.length];
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioSum += mParameters.mThreadsRatio[i];
+      }
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioArray[i] = last + (mParameters.mThreadsRatio[i] / ratioSum) * mParameters.mThreads;
+        last = ratioArray[i];
+      }
+      int operationIndex = 0;
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        if (ratioArray[i] >= mThreadIndex) {
+          operationIndex = i;
+          break;
+        }
+      }
```

Comment:
consider dedup the code with above

Reply:
yup i refactored it into a method with comments